concent_api/core/validation.py

```diff
@@ -434,3 +430,33 @@ def validate_frames(frames_list: List[int]):
                 'Frame number must be grater than 0',
                 ErrorCode.MESSAGE_FRAME_VALUE_NOT_POSITIVE_INTEGER
             )
+
+
+def validate_compute_task_def(compute_task_def: message.tasks.ComputeTaskDef) -> None:
+    string_fields = ["output_format", "scene_file"]
+    integer_fields = ["frames"]
+
+    validate_int_value(compute_task_def['deadline'])
+
+    validate_id_value(compute_task_def['task_id'], 'task_id')
+    validate_id_value(compute_task_def['subtask_id'], 'subtask_id')
+
+    extra_data = compute_task_def.get("extra_data")
+    if extra_data is None:
+        raise MessageValidationError(
+            "'extra_data' is missing in ComputeTaskDef",
+            ErrorCode.MESSAGE_INVALID
+        )
+    for mandatory_data in string_fields + integer_fields:
+        if mandatory_data not in extra_data:
+            raise MessageValidationError(
+                f"{mandatory_data} is missing in ComputeTaskDef",
+                ErrorCode.MESSAGE_INVALID
+            )
+    validate_frames(extra_data["frames"])
```

Comment:
Here it should have separate line before and after this call.

Reply:
done